protocol/contracts/TrucoMatch.sol

```diff
@@ -10,42 +10,132 @@ import "./crypt/DeckCrypt.sol";
 
 contract TrucoMatch  {
 
+    struct player {
+        address playerAddress;
+        uint256 tokensAtStake;
+    }
+    
     struct Match {
-        address[2] players; // player 0 is the creator of the match
+        player[2] players; // player 0 is the creator of the match
         CardsStructs.Deck deck;
         CardsStructs.GameState gameState;
         uint256 tokensAtStake; 
+
     }
 
     IERC3333 trucoEngine;
     DeckCrypt deckCrypt;
-    IERC20 trucoin;
+    IERC20 truCoin;
     Match currentMatch;
 
-    constructor (IERC3333 _trucoEngine, DeckCrypt _deckCrypt, IERC20 _trucoin) {
+    // Events
+    event MatchCreated(address match_address);
+    event MatchStarted(address player1, address player2, uint256 tokensAtStake);
+    event PlayerStaked(address player, uint256 tokensAtStake);
```

Comment:
I think the correct wording for this action in the context of the game should be "Bet", instead of "stake".